srtp/srtp.c

```diff
@@ -769,13 +769,13 @@ static inline int base_key_length(const srtp_cipher_type_t *cipher, int key_leng
   case SRTP_AES_ICM_256:
     /* The legacy modes are derived from
      * the configured key length on the policy */
-    return key_length - 14;
+    return key_length - SRTP_SALT_LEN;
```

Comment:
The reason it exists is that the key and salt are assumed to be concatenated and there are a few places where the two need to be split.  Had the key and salt been separate fields, this function would not be needed.  And the function is useful to avoid a number switch statements in other places.

I don't have a really strong preference, though I do have a slight one toward reducing the number of computations.  That said, this is only invoked when initializing keys and that doesn't happen frequently, so it's not a big deal.